lib/barbeque/message_queue.rb

```diff
@@ -23,9 +25,22 @@ def dequeue
         valid_messages.concat(messages)
       end
 
-      valid_messages.shift.tap do |message|
-        delete_message(message)
+      message = valid_messages.shift
+      # XXX: #receive_messages returns one message at most because it calls
+      #       receive_message API without max_number_of_messages option.
+      unless valid_messages.empty?
+        extend_visibility_timeout(valid_messages)
```

Comment:
Why do you extend VisibilityTimeout here instead of changing SQS queue's VisibilityTimeout to 60?